marine.c

```diff
@@ -633,7 +634,50 @@ WS_DLL_PUBLIC int validate_display_filter(char *dfilter) {
     return TRUE;
 }
 
-WS_DLL_PUBLIC int marine_add_filter(char *bpf, char *dfilter, char **fields, int fields_len, char *err_msg) {
+int parse_output_fields(output_fields_t *output_fields, char **fields, int fields_len, char **err_msg) {
+    for (int i = 0; i < fields_len; i++) {
+        output_fields_add(output_fields, fields[i]);
+    }
+
+    GSList *it = NULL;
+    GSList *invalid_fields = output_fields_valid(output_fields);
+    if (invalid_fields != NULL) {
+        if (err_msg == NULL) {
+            return -1;
+        }
+
+        int current_err_size = 64;
+        *err_msg = (char *)malloc(current_err_size);
+        strcpy(*err_msg, "Some fields aren't valid:\n");
+
+        for (it = invalid_fields; it != NULL; it = g_slist_next(it)) {
+            current_err_size = current_err_size + strlen((gchar *) it->data) + 1;
+            *err_msg = (char *) realloc(*err_msg, current_err_size);
+            strcat(*err_msg, "\t");
+            strcat(*err_msg, (gchar *) it->data);
+        }
+
+        output_fields_free(output_fields);
+        g_slist_free(invalid_fields);
+
+        return -1;
+    }
+
+    return 0;
+}
+
+WS_DLL_PUBLIC int validate_fields(char **fields, int fields_len) {
```

Comment:
As for the API - I think it will be interesting to tell the user which fields are invalid.
@yehonatanz ?

Reply:
I think that's a good idea, but how would you expose it in python?
Raising an error would break the compatibility with the other validation functions.

Reply:
Return a tuple like the filter and parse function maybe?

Reply:
If we do that shouldn't we also make 'validate_display_filters` return it's error?
It can also be pretty informative.

Reply:
Returning a tuple could be annoying when you just want to `if validate_fields(...):`.
I agree with @shaharyarn about `validate_display_filter`, the API should be consistent.
For now let's just leave the error message, and add a separate issue for it.